glommio/src/sys/mod.rs

```diff
@@ -365,6 +365,7 @@ pub(crate) enum SourceType {
     Statx(CString, Box<RefCell<libc::statx>>),
     Timeout(TimeSpec64),
     Connect(SockAddr),
+    ConnectTimeout(SockAddr, TimeSpec64),
```

Comment:
This is unrelated, but FYI: I. have a pending patch that removes `SockAddr` from `SourceType` and replaces it with `Box<SockAddr>`. Those larger fields are bloating the size of `SourceType` for everybody.

No action needed now, but keep an eye out for changes in the near future as this may conflict with this change.